src/sentry/search/events/builder/base.py

```diff
@@ -570,7 +573,8 @@ def resolve_params(self) -> list[WhereType]:
         # complain on an empty list which results on no data being returned.
         # This change will prevent calling Snuba when no projects are selected.
         # Snuba will complain with UnqualifiedQueryError: validation failed for entity...
-        if not self.params.project_ids:
+        project_ids = self.resolve_projects()
+        if not project_ids:
```

Comment:
Could this result in poor UX? For example, users go on to a page like discover before sending an event then they get this error saying they need to select an project when that problem is that they need to send an event first.

Reply:
🤔 I think I'll just add `You need to select at least one project with data`